### DISCUSSION TITLE
Select locale by cookie/localStorage

### DISCUSSION THREAD

**Discussion by jim-king-2000** [amannn/next-intl#441] [Q&A]
### Is your feature request related to a problem? Please describe.

I check the example and find that it selects the locale by url. `https://xxx.com/en/` is for English and `https://xxx.com/de/` for German.

### Describe the solution you'd like

Could you please provide an example that it selects locale by cookie/localStorage instead of url?

### Describe alternatives you've considered

We don't use additional path of url to select locale. It destroy the original structure of the existing project. We'd like to resort to cookie/localStorage.

**Comment by amannn** [comment]
I think you're looking for this: https://next-intl-docs.vercel.app/docs/routing/middleware#never-use-a-locale-prefix

I'll move this to to a discussion since it's a usage question.

**Reply by jim-king-2000** [reply]
Not exactly, but I'll investigate it. Thank you.
